nibabel/streamlines/tractogram.py

```diff
@@ -0,0 +1,543 @@
+            points for a particular streamline t and M is the number of
+            scalars associated to each point (excluding the three
+            coordinates).
+        """
+        self.streamlines = streamlines
+        self.data_per_streamline = data_per_streamline
+        self.data_per_point = data_per_point
+        self._affine_to_rasmm = np.eye(4)
+
+    @property
+    def streamlines(self):
+        return self._streamlines
+
+    @streamlines.setter
+    def streamlines(self, value):
+        self._streamlines = ArraySequence(value)
+
+    @property
+    def data_per_streamline(self):
+        return self._data_per_streamline
+
+    @data_per_streamline.setter
+    def data_per_streamline(self, value):
+        self._data_per_streamline = Tractogram.DataPerStreamlineDict(self,
+                                                                     value)
+
+    @property
+    def data_per_point(self):
+        return self._data_per_point
+
+    @data_per_point.setter
+    def data_per_point(self, value):
+        self._data_per_point = Tractogram.DataPerPointDict(self, value)
+
+    @property
+    def affine_to_rasmm(self):
+        # Return a copy. User should use self.apply_affine` to modify it.
+        return self._affine_to_rasmm.copy()
+
+    def __iter__(self):
+        for i in range(len(self.streamlines)):
+            yield self[i]
+
+    def __getitem__(self, idx):
+        pts = self.streamlines[idx]
+
+        data_per_streamline = {}
+        for key in self.data_per_streamline:
+            data_per_streamline[key] = self.data_per_streamline[key][idx]
+
+        data_per_point = {}
+        for key in self.data_per_point:
+            data_per_point[key] = self.data_per_point[key][idx]
+
+        if isinstance(idx, (int, np.integer)):
+            return TractogramItem(pts, data_per_streamline, data_per_point)
+
+        return Tractogram(pts, data_per_streamline, data_per_point)
+
+    def __len__(self):
+        return len(self.streamlines)
+
+    def copy(self):
+        """ Returns a copy of this :class:`Tractogram` object. """
+        data_per_streamline = {}
+        for key in self.data_per_streamline:
+            data_per_streamline[key] = self.data_per_streamline[key].copy()
+
+        data_per_point = {}
+        for key in self.data_per_point:
+                data_per_point[key] = self.data_per_point[key].copy()
+
+        tractogram = Tractogram(self.streamlines.copy(),
+                                data_per_streamline,
+                                data_per_point)
+
+        tractogram._affine_to_rasmm = self.affine_to_rasmm
+        return tractogram
+
+    def apply_affine(self, affine, lazy=False):
+        """ Applies an affine transformation on the points of each streamline.
+
+        If `lazy` is not specified, this is performed *in-place*.
+
+        Parameters
+        ----------
+        affine : ndarray shape (4, 4)
```

Comment:
`to_rasmm` or `to_world` sounds like a reasonable idea.

Would you mind adding an explanation like the text above to the class docstring to be clear what the coordinates of the streamlines mean (I guess you could phrase that as "any coordinate meaning of your choice, where `affine_to_rasmm` brings the coordinates into RAS+ space, and where `affine_to_rasmm` can be the identity (`np.eye(4)`) or None (transformation unknown)."  Is that right?